src/shared/dissect-image.c

```diff
@@ -1661,6 +1698,40 @@ int dissect_image_and_warn(
         }
 }
 
+bool dissected_image_can_do_verity(const DissectedImage *image, unsigned partition_number) {
+        int k;
+
+        k = PARTITION_VERITY_OF(partition_number);
+        if (k >= 0)
+                return true;
+        if (image->can_verity) {
+                /* image->can_verity is meaningful on a per-partition check only with a single file system */
+                for (unsigned i = PARTITION_ROOT; i < _PARTITION_DESIGNATOR_MAX; i++)
+                        if (image->partitions[i].found)
+                                return false;
```

Comment:
this loop will always return false, since if there's no part table we detect the whole image as `PARTITION_ROOT` designator

Reply:
Forgot the ```+ 1``` in the initialization - fixed